internal/services/steelthreadtesting/steelthread_test.go

```diff
@@ -0,0 +1,127 @@
+//go:build steelthread
+// +build steelthread
+
+package steelthreadtesting
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"os"
+	"testing"
+	"time"
+
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+	"google.golang.org/grpc/test/bufconn"
+	yamlv3 "gopkg.in/yaml.v3"
+
+	v1 "github.com/authzed/authzed-go/proto/authzed/api/v1"
+	"github.com/authzed/spicedb/internal/datastore/memdb"
+	"github.com/authzed/spicedb/internal/dispatch/graph"
+	"github.com/authzed/spicedb/internal/grpchelpers"
+	"github.com/authzed/spicedb/internal/middleware/consistency"
+	datastoremw "github.com/authzed/spicedb/internal/middleware/datastore"
+	v1svc "github.com/authzed/spicedb/internal/services/v1"
+	"github.com/authzed/spicedb/pkg/validationfile"
+	"github.com/dustin/go-humanize"
+	"github.com/gosimple/slug"
+	"github.com/stretchr/testify/require"
+)
+
+const defaultConnBufferSize = humanize.MiByte
+
+func TestSteelThreads(t *testing.T) {
+	for _, tc := range steelThreadTestCases {
+		t.Run(tc.name, func(t *testing.T) {
+			ctx, cancel := context.WithTimeout(context.Background(), 60*time.Second)
+			defer cancel()
+
+			// Create a new datastore.
+			ds, err := memdb.NewMemdbDatastore(0, 0*time.Second, memdb.DisableGC)
```

Comment:
I'm wondering about the choice of memdb here. We could definitely have bugs that would be datastore-specific and affect a steelthread test, right? Should we run against all backends?

Reply:
Yes, that was my plan as a followup